renderdoc/driver/vulkan/vk_win32.cpp

```diff
@@ -101,6 +101,17 @@ void WrappedVulkan::AddRequiredExtensions(bool instance, vector<string> &extensi
                    VK_KHR_WIN32_SURFACE_EXTENSION_NAME) == extensionList.end())
         extensionList.push_back(VK_KHR_WIN32_SURFACE_EXTENSION_NAME);
     }
+
+    if (supportedExtensions.find(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME) == supportedExtensions.end())
+    {
+      RDCWARN("Unsupported required instance extension for AMD performance counters '%s'", VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+    }
+    else
+    {
+      if (std::find(extensionList.begin(), extensionList.end(),
+        VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME) == extensionList.end())
+        extensionList.push_back(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+    }
```

Comment:
Since this is needed on both platforms it should instead be done outside these files in `vk_device_funcs.cpp` in `WrappedVulkan::Serialise_vkCreateDevice`, the same place where `EXT_debug_marker` and `AMD_shader_info` are enabled.

Also as a follow up, do we need to store a flag somewhere as to whether this extension was enabled? So we can check it later to gate any use that might depend on it. I don't see anything directly so I assume it's inside the GPA libraries themselves?

Reply:
Fixed.  It does not need to be saved.  If not found, it informs the user why they won't see any performance counters.

Reply:
My concern is that most users will not look at a warning printed to the log file, or won't notice it in amongst other log prints. I think we should at least save this as a bool and check it before enumerating AMD counters (i.e. treat the situation where the extension is missing as the same as when the GPA interface fails to initialise). That way users might wonder why the AMD counters aren't available but they at least won't be shown counters that then don't work.

Reply:
To clarify, the counters won't be populated.  This is the user's log file info as to why they weren't populated.

Reply:
Oh OK, sorry I misunderstood, that's fine then!